collectors/plugins.d/plugins_d.c

```diff
@@ -246,6 +277,115 @@ inline size_t pluginsd_process(RRDHOST *host, struct plugind *cd, FILE *fp, int
+    static RRDSET   *__stream_prs_st = NULL;     //       for active processors
+    static RRDSET   *__stream_metrics_st = NULL; //       for active metrics we collect from slaves
+    static RRDSET   *__stream_processes_st = NULL;  //    for total active processes in slaves
+    static time_t   stream_updated   = 0;           // Used so that we Do not rrdset_done in less than a second
+    time_t          thread_start_time;
+    static int      stream_stats_activated = 0;
+    
+	if (likely(stream_stats_enabled) && likely(!stream_stats_activated) && likely(host != localhost)) {
+        stream_stats_activated = 1;
+
+        __stream_mem_st = rrdset_create_localhost(
+                "netdata"
+                , "memory" 
+                , NULL
+                , "streams"
+                , NULL
+                , "NetData Incoming Streams"
+                , "MB"
+                , "netdata"
+                , "stats"
+                , 130000
+                , localhost->rrd_update_every
+                , RRDSET_TYPE_LINE
+            );
+        __stream_cnt_st = rrdset_create_localhost(
+                "netdata"
+                , "stream count" 
+                , "streams"
+                , "streams"
+                , NULL
+                , "NetData Incoming Streams"
+                , "Count"
+                , "netdata"
+                , "stats"
+                , 130001
+                , localhost->rrd_update_every
+                , RRDSET_TYPE_LINE
+            );
+            __stream_prs_st = rrdset_create_localhost(
+                "netdata"
+                , "active processors" 
+                , "streams"
+                , "streams"
+                , NULL
+                , "NetData Incoming Streams"
+                , "Count"
+                , "netdata"
+                , "stats"
+                , 130002
+                , localhost->rrd_update_every
+                , RRDSET_TYPE_LINE
+            );
+            __stream_metrics_st = rrdset_create_localhost(
+                "netdata"
+                , "metrics" 
+                , "streams"
+                , "streams"
+                , NULL
+                , "NetData Incoming Streams"
+                , "Count"
+                , "netdata"
+                , "stats"
+                , 130003
+                , localhost->rrd_update_every
+                , RRDSET_TYPE_LINE
+            );
+            __stream_processes_st = rrdset_create_localhost(
+                "netdata"
+                , "processes" 
+                , "streams"
+                , "streams"
+                , NULL
+                , "NetData Incoming Streams"
+                , "Count"
+                , "netdata"
+                , "stats"
+                , 130004
+                , localhost->rrd_update_every
+                , RRDSET_TYPE_LINE
+            );
+
+            if (likely(__stream_mem_st)) 
```

Comment:
Hi @stelfrag,

Please, add an else in this group for you write an error log case there was a problem to allocate the variables. I saw that you are checking whether these variables are not null in the rest of the code, but it is also good to know when something wrong happens.

Reply:
Thank you @thiagoftsm for pointing this out!! I will add it

Reply:
Nice! I will  mark as resolved, because you solved as expected! :)